vj4/handler/problem.py

```diff
@@ -523,10 +551,16 @@ class ProblemDataHandler(base.Handler):
     # domain administrators will have PERM_READ_PROBLEM_DATA,
     # problem owner will have PERM_READ_PROBLEM_DATA_SELF.
     pdoc = await problem.get(self.domain_id, pid)
+    ddoc, dudoc = self.domain, self.domain_user
```

Comment:
A simpler but slower way is to redirect to target `/d/xxx/p/yyy/data`.
Please discuss.